tests/test_dom0_and_templates_correct_key.py

```diff
@@ -0,0 +1,248 @@
+        "yum_repo_url": "https://yum-test.securedrop.org/workstation/dom0/$FEDORA_VERSION/",
+    },
+}
+
+
+# Get expected configuration (environment-dependent)
+def _get_expected_config():
+    with open("config.json") as c:
+        config = json.load(c)
+        env = config.get("environment")
+
+        # Fall back to checking environment via keyring package
+        # (will eventually replace config.json environment)
+        if not env:
+            env = _get_env_by_package()
+
+    config = REPO_CONFIG[env].copy()
+
+    # Fedora version is a placeholder, so fix that
+    config["yum_repo_url"] = Template(config["yum_repo_url"]).safe_substitute(
+        {"FEDORA_VERSION": FEDORA_VERSION}
+    )
+    return env, config
+
+
+def _get_env_by_package():
+    """
+    Check which environment we are using by checking keyring package.
+    Order matters; due to package versioning, an installed dev package
+    means a dev setup, so check dev, staging, then prod.
+    """
+    for env, suffix in [("dev", "-dev"), ("staging", "-staging"), ("prod", "")]:
+        if _is_installed(f"securedrop-workstation-keyring{suffix}"):
+            return env
+    pytest.fail("No keyring package")
+    # Unreachable, but ruff was unhappy
+    return None
+
+
+def _is_installed(pkg: str):
+    """
+    Check if package is installed.
+    """
+    try:
+        subprocess.check_call(
+            ["rpm", "-q", pkg], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+        )
+
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+# A bit clunky, but we have limited tooling options in dom0.
+def _read_gpg_key_details(key_path: str):
+    """# Use system-installed gpg to read pubkey details."""
+    if not os.path.isfile(key_path):
+        raise FileNotFoundError(f"No such file: {key_path}")
+
+    with tempfile.TemporaryDirectory() as temp_home:
+        subprocess.check_call(
+            ["gpg2", "--homedir", temp_home, "--import", "--quiet", "--batch", "--yes", key_path],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+
+        output = subprocess.check_output(
+            ["gpg2", "--homedir", temp_home, "--with-colons", "--list-keys"]
+        ).decode()
+
+        return _parse_gpg_output(output)
+
+
+def _parse_gpg_output(output: str):
+    key_info = {"fingerprint": None, "uids": [], "expires": None}
+
+    for line in output.splitlines():
+        parts = line.strip().split(":")
+        tag = parts[0]
+
+        if tag == "fpr" and not key_info["fingerprint"]:
+            key_info["fingerprint"] = parts[9]
+        elif tag == "pub":
+            exp = parts[6]
+            if exp:
+                key_info["expires"] = exp
+        elif tag == "uid":
+            key_info["uids"].append(parts[9])
+
+    return key_info
+
+
+def _validate_key(info: dict, expected_fpr: str, expected_uid: str, expected_expiry: str):
```

Comment:
instead of checking things and raising exceptions, I think we can just use normal pytest assertions